indexing-service/src/main/java/io/druid/indexing/common/task/RealtimeIndexTask.java

```diff
@@ -293,35 +292,40 @@ public String getVersion(final Interval interval)
       long nextFlush = new DateTime().plus(intermediatePersistPeriod).getMillis();
       while (firehose.hasMore()) {
         final InputRow inputRow;
+
         try {
           inputRow = firehose.nextRow();
+
           if (inputRow == null) {
+            log.debug("thrown away null input row, considering unparseable");
+            fireDepartment.getMetrics().incrementUnparseable();
             continue;
           }
+        }
+        catch (ParseException e) {
+          log.debug(e, "thrown away line due to exception, considering unparseable");
+          fireDepartment.getMetrics().incrementUnparseable();
+          continue;
+        }
 
-          int currCount = plumber.add(inputRow);
-          if (currCount == -1) {
-            fireDepartment.getMetrics().incrementThrownAway();
-            log.debug("Throwing away event[%s]", inputRow);
-
-            if (System.currentTimeMillis() > nextFlush) {
-              plumber.persist(firehose.commit());
-              nextFlush = new DateTime().plus(intermediatePersistPeriod).getMillis();
-            }
+        int currCount = plumber.add(inputRow);
+        if (currCount == -1) {
+          fireDepartment.getMetrics().incrementThrownAway();
+          log.debug("Throwing away event[%s]", inputRow);
 
-            continue;
-          }
-
-          fireDepartment.getMetrics().incrementProcessed();
-          final Sink sink = plumber.getSink(inputRow.getTimestampFromEpoch());
-          if ((sink != null && !sink.canAppendRow()) || System.currentTimeMillis() > nextFlush) {
+          if (System.currentTimeMillis() > nextFlush) {
             plumber.persist(firehose.commit());
             nextFlush = new DateTime().plus(intermediatePersistPeriod).getMillis();
           }
+
+          continue;
```

Comment:
No, because there's a continue; after a null inputRow is found, so the loop will go directly back to firehose.hasMore() + firehose.nextRow() before trying to persist anything.